File: solutions/elixir/resistor-color-duo/1/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do

  @resistor_color %{
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }

  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value(colors) do
    [first, second | _] = colors

    @resistor_color[first] * 10 + @resistor_color[second]
  end
end

File: solutions/lua/resistor-color-duo/1/resistor-color-duo.lua
local colors = {
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
}
function color_code(color)
    for i, col in ipairs(colors) do
        if col == color then return i - 1 end
    end
end

return {
  value = function(colors)
    return color_code(colors[1]) * 10 + color_code(colors[2])
  end
}
